xwiki-platform-core/xwiki-platform-dashboard/xwiki-platform-dashboard-ui/src/main/resources/Dashboard/XWikiUserDashboardSheet.xml

```diff
@@ -283,7 +283,12 @@
       &lt;div class="full column xform"&gt;
         &lt;div class="userDashboard"&gt;
           #if($xcontext.action == 'view' &amp;&amp; $hasEdit)
-            &lt;div class="editProfileCategory"&gt;&lt;a href="$doc.getURL('edit', 'category=dashboard')"&gt;&lt;span class="hidden"&gt;$services.localization.render('platform.core.profile.category.dashboard.edit')&lt;/span&gt;&lt;/a&gt;&lt;/div&gt;
+            &lt;div class="editProfileCategory"&gt;
+              &lt;a href="$doc.getURL('edit', 'category=dashboard')" class="btn btn-xs"&gt;
+                &lt;span class="action-icon"&gt;$services.icon.renderHTML('pencil')&lt;/span&gt;
+                &lt;span class='hidden'&gt;$escapetool.xml($services.localization.render('platform.core.profile.category.dashboard.edit'))&lt;/span&gt;
+              &lt;/a&gt;
+            &lt;/div&gt;
```

Comment:
Thanks @slauriere for the PR! About the new code layout, did you check the resulting HTML to see if this didn't add new lines and empty spaces? I think it could have been the reason why it was done that way.

I also like the way you did it, it's much simpler to read and I'd love to standardize on this. I think we need a preprocessor to remove empty lines and spaces though. Note: I see it's inside a velocity macro which means it'll remove the leading spaces; there'll still be the extra new lines probably though.

WDYT? Also curious to know what others think about it.

Reply:
Thanks @vmassol. I confirm the resulting HTML does not contain new lines nor empty spaces, at least when not using a screen reader, here's a capture in Firefox:

![profile-1](https://user-images.githubusercontent.com/1022195/126181371-2971adb7-189b-492f-abbf-173c1fbf592e.jpg)

However removing the "sr-only" class to simulate what happens on screen readers generates a whitespace (because we have now two visible spans next to another):

![profile-2](https://user-images.githubusercontent.com/1022195/126181394-193e4205-f4fe-4df3-bd41-afc27054ab0c.jpg)

I don't know how this whitespace is interpreted by screen readers. Since the leading span contains no text, it may be considered as the first whitespace of a text, hence ignored? Not sure. Another approach would consist in using the `aria-label` attribute instead of `sr-only` CSS class to provide the button purpose. Both approaches seem to be fine, I couldn't find whether one is recommanded over the other so far. On [this accessibility recommendation by Font Awesome](https://fontawesome.com/v5.15/how-to-use/on-the-web/other-topics/accessibility), the two spans are not juxtaposed:

```
<i aria-hidden="true" class="fas fa-car" title="Time to destination by car"></i>
<span class="sr-only">Time to destination by car:</span>
<span>4 minutes</span>
```

For insteractive elements such as a link in our case, the Font Awesome documentation suggests using `aria-label` rather than `sr-only`. However [this article](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/) states that "the [aria-label] attribute isn’t announced consistently by web browsers".

Note that using `aria-hidden` seems to be recommended with font icons, I'll create an issue about that.

As for new lines, I have this naive question: even if the code were not in a Velocity macro, can new lines actually get generated by the HTML macro even with `wiki="true"`?

Reply:
I wouldn't trust the HTML view tab in FF because it'll possibly remove whitespace and empty new lines. You should check the raw source I think.

Reply:
> I don't know how this whitespace is interpreted by screen readers. 

Best is to test it I guess :)

Reply:
> As for new lines, I have this naive question: even if the code were not in a Velocity macro, can new lines actually get generated by the HTML macro even with wiki="true"?

AFAIK only the Velocity macro has a filter to remove leading spaces. See https://extensions.xwiki.org/xwiki/bin/view/Extension/Velocity%20Macro#HParametersdefinition

Reply:
Re `aria-label` vs `sr-only`, I'm clueless, I haven't researched the topic. It's important that you make proposals on the forum since we don't have any decisions on anything beyond the automated Dutch WCAG guidelines tests.